src/Stack/Options/ExecParser.hs

```diff
@@ -70,3 +71,9 @@ execOptsExtraParser = eoPlainParser <|>
     eoPlainParser = flag' ExecOptsPlain
                           (long "plain" <>
                            help "Use an unmodified environment (only useful with Docker)")
+
+    eoCwdParser :: Parser (Maybe FilePath)
+    eoCwdParser = optional
+                  (strOption (long "cwd"
```

Comment:
Please also add 

```
               metavar "DIR" <>
               completer dirCompleter <>
```

For better help output and path completion for this option.

Reply:
Done :)